src/hid/parser.py

```diff
@@ -12,10 +12,11 @@
 
 class HidParserError(HidUsageError):
     """Failed to parse a line in file"""
-    def __init__(self, filepath, line):
-        super().__init__('HID usage parsing error in file ' + filename + ' (line="' + line + '")')
-    def __init__(self, page):
-        super().__init__('HID usage parsing error in page ' + page.name())
+    def __init__(self, filepath=None, line=None, page=None):
+        if (filepath is not None):
+            super().__init__('HID usage parsing error in file ' + filepath + ' (line="' + line + '")')
+        else:
+            super().__init__('HID usage parsing error in page ' + page.name())
```

Comment:
Could you please leave this constructor intact, and create an additional one that only takes the page parameter instead?